libraries/apollo-ast/src/main/kotlin/com/apollographql/apollo3/ast/check_key_fields.kt

```diff
@@ -92,23 +92,23 @@ private fun CheckKeyFieldsScope.collectFields(
           return@flatMap emptyList()
         }
 
-        collectFields(it.selectionSet.selections, it.typeCondition.name, implementedTypes)
+        collectFields(it.selections, it.typeCondition?.name ?: parentType, implementedTypes)
```

Comment:
This is the other breaking change. Our antlr parser didn't support inline fragments without a type condition. See https://spec.graphql.org/draft/#sel-GAFbfLABAB_F3kG:

```graphql
query inlineFragmentNoType($expandedInfo: Boolean) {
  user(handle: "zuck") {
    id
    name
    ... @include(if: $expandedInfo) {
      firstName
      lastName
      birthday
    }
  }
}
```

This is now supported by the parser but I'm considering throwing in the compiler because it makes the naming of fragment classes a bit awkward. Any thoughts here?

Reply:
Unless I missed it, I think we never got any ask or complaint from users wanting to do this? (Although that doesn't necessary mean nobody encountered it, but maybe the workaround is easy enough that nobody reached out).

In addition to `@include`, one place where this can be nice is to use `@defer`.

But for the name... `fragment1`, `fragment2`, etc.?

In any case I find it reasonable to keep it unsupported and at least it's cool that with this new parser we have the ability to support it, now or later.

Reply:
> we never got any ask or complaint from users wanting to do this

Exactly. "Don't fix it if it ain't broke"

> one place where this can be nice is to use `@defer`

Indeed. Well, it's always easier to add than remove so I'll throw for now and if we get requests for this we can always add it.

Reply:
check added in https://github.com/apollographql/apollo-kotlin/pull/5047/commits/84e30bad7b648f1648bb8f4e664afcfd347f73ea